lib/oelite/function.py

```diff
@@ -120,6 +121,28 @@ def __init__(self, meta, var, name=None, tmpdir=None):
         super(ShellFunction, self).__init__(meta, var, name, tmpdir)
         return
 
+    def runscript(self, cmd):
+        cmdstr = cmd
+        cmdname = cmd.split(None, 1)[0]
+
+        print '> %s'%(cmdstr,)
+
+        try:
+            retval = None
+            returncode = subprocess.call(cmd, stdin=sys.stdin, shell=True)
+            if returncode == 0:
+                retval = True
+            else:
+                print "Error: Command failed: %r: %d"%(cmdstr, returncode)
+        except OSError, e:
+            if e.errno == 2:
+                print "Error: Command not found:", cmdname
+            else:
```

Comment:
Yes, but that's quite deliberate. We just want to return False when the command fails. Please look at util.shcmd where this is copied (and simplified) from. This commit is supposed to be a semantic noop.